cmd/gcp-controller-manager/kubelet_readonly_csr_approver.go

```diff
@@ -252,22 +254,57 @@ func validatePodAnnotation(request kubeletReadonlyCSRRequest) kubeletReadonlyCSR
 		klog.Errorf("csr does not have an attached pod name")
 		return kubeletReadonlyCSRResponse{
 			result:  false,
-			err:     fmt.Errorf("csr does not have an attached pod name"),
+			err:     nil,
 			message: "csr does not have pod name attached",
 		}
 	}
 
+	podUIDs, ok := request.csr.Spec.Extra[podUIDKey]
+	if !ok {
+		klog.Errorf("csr does not have an attached pod UID")
+		return kubeletReadonlyCSRResponse{
+			result:  false,
+			err:     nil,
+			message: "csr does not have pod UID attached",
+		}
+	}
+
+	if len(podNames) != len(podUIDs) {
+		return kubeletReadonlyCSRResponse{
+			result:  false,
+			err:     nil,
+			message: "bad request: length of pod name and UID are not equal",
+		}
+	}
+
+	username := request.csr.Spec.Username
+	if len(username) == 0 {
+		return kubeletReadonlyCSRResponse{
+			result:  false,
+			err:     nil,
+			message: "username is missing",
+		}
+	}
+	namespace, _, err := serviceaccount.SplitUsername(username)
```

Comment:
add a comment here saying we are getting the namespace from the username and also provide an example.